File: app/src/main/java/com/example/ui/FuturePathViewModel.kt
package com.example.ui

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.viewModelScope
import com.example.data.*
import kotlinx.coroutines.flow.*
import kotlinx.coroutines.launch

class FuturePathViewModel(application: Application) : AndroidViewModel(application) {
    private val repository = CareerRepository(application)

    // UI States
    val userProfile: StateFlow<UserProfile> = repository.getUserProfile()
        .map { it ?: UserProfile() }
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), UserProfile())

    val chatMessages: StateFlow<List<ChatMessage>> = repository.getChatMessages()
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), emptyList())

    val simulation: StateFlow<CareerSimulation?> = repository.getSimulation()
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), null)

    // Screen navigation tracking
    private val _currentScreen = MutableStateFlow("splash")
    val currentScreen: StateFlow<String> = _currentScreen.asStateFlow()

    // Interactive UI controls
    private val _isSimulating = MutableStateFlow(false)
    val isSimulating: StateFlow<Boolean> = _isSimulating.asStateFlow()

    private val _isChatLoading = MutableStateFlow(false)
    val isChatLoading: StateFlow<Boolean> = _isChatLoading.asStateFlow()

    private val _simulationComplete = MutableStateFlow(false)
    val simulationComplete: StateFlow<Boolean> = _simulationComplete.asStateFlow()

    // Form states temporary hold
    private val _formAge = MutableStateFlow(17)
    val formAge = _formAge.asStateFlow()

    private val _formCountry = MutableStateFlow("Nigeria")
    val formCountry = _formCountry.asStateFlow()

    private val _formInterests = MutableStateFlow("Technology, Coding, Mathematics")
    val formInterests = _formInterests.asStateFlow()

    private val _formSkills = MutableStateFlow("Basic Logic, English, Writing")
    val formSkills = _formSkills.asStateFlow()

    private val _formGrades = MutableStateFlow("A in Math, B in English, A in Physics")
    val formGrades = _formGrades.asStateFlow()

    private val _formBudget = MutableStateFlow("Under $1,000/year (Requires scholarships/free paths)")
    val formBudget = _formBudget.asStateFlow()

    private val _formInternet = MutableStateFlow("Mobile Data only, capped (Low Bandwidth)")
    val formInternet = _formInternet.asStateFlow()

    private val _formLifestyle = MutableStateFlow("Remote Freelancer or Tech Entrepreneur")
    val formLifestyle = _formLifestyle.asStateFlow()

    // Initializer
    init {
        // Prepare database with default items on launch
        viewModelScope.launch {
            val existing = repository.getUserProfile().firstOrNull()
            if (existing == null) {
                repository.saveProfile(UserProfile())
            }
        }
    }

    // Setters
    fun navigateTo(screen: String) {
        _currentScreen.value = screen
    }

    fun updateFormFields(
        age: Int,
        country: String,
        interests: String,
        skills: String,
        grades: String,
        budget: String,
        internet: String,
        lifestyle: String
    ) {
        _formAge.value = age
        _formCountry.value = country
        _formInterests.value = interests
        _formSkills.value = skills
        _formGrades.value = grades
        _formBudget.value = budget
        _formInternet.value = internet
        _formLifestyle.value = lifestyle
    }

    fun saveUserProfileAndRunSimulation() {
        viewModelScope.launch {
            _isSimulating.value = true
            val profile = UserProfile(
                age = _formAge.value,
                country = _formCountry.value,
                interests = _formInterests.value,
                skills = _formSkills.value,
                grades = _formGrades.value,
                budget = _formBudget.value,
                internetAccess = _formInternet.value,
                preferredLifestyle = _formLifestyle.value,
                xp = userProfile.value.xp,
                level = userProfile.value.level,
                streak = userProfile.value.streak,
                futureScore = userProfile.value.futureScore,
                isPremium = userProfile.value.isPremium
            )
            repository.saveProfile(profile)
            repository.runCareerSimulation(profile)
            _isSimulating.value = false
            _simulationComplete.value = true
            _currentScreen.value = "dashboard"
        }
    }

    fun togglePremium(isPremium: Boolean) {
        viewModelScope.launch {
            val prof = userProfile.value.copy(isPremium = isPremium)
            repository.saveProfile(prof)
        }
    }

    fun sendChatMessage(text: String) {
        if (text.isBlank()) return
        viewModelScope.launch {
            _isChatLoading.value = true
            repository.sendMessage("user", text)
            _isChatLoading.value = false
        }
    }

    fun clearChatHistory() {
        viewModelScope.launch {
            repository.clearChat()
        }
    }

    // Award bonus event XP for dashboard learning engagement clicks
    fun claimDailyReward() {
        viewModelScope.launch {
            val profile = userProfile.value
            val bonus = 25
            val currentXp = profile.xp + bonus
            val level = 1 + (currentXp / 300)
            repository.saveProfile(profile.copy(
                xp = currentXp,
                level = level,
                streak = profile.streak + 1,
                lastUpdate = System.currentTimeMillis()
            ))
        }
    }
}

File: lib/viewmodel/futurepath_viewmodel.dart
import 'package:flutter/foundation.dart';
import '../data/models.dart';
import '../data/repository.dart';

/// Mirrors FuturePathViewModel.kt — ChangeNotifier replaces StateFlow
class FuturePathViewModel extends ChangeNotifier {
  final _repo = CareerRepository();

  // ── State ──────────────────────────────────────────────────────
  UserProfile _userProfile = const UserProfile();
  List<ChatMessage> _chatMessages = [];
  CareerSimulation? _simulation;
  String _currentScreen = 'splash';
  bool _isSimulating = false;
  bool _isChatLoading = false;
  bool _simulationComplete = false;

  // Form fields
  int _formAge = 17;
  String _formCountry = 'Nigeria';
  String _formInterests = 'Technology, Coding, Mathematics';
  String _formSkills = 'Basic Logic, English, Writing';
  String _formGrades = 'A in Math, B in English, A in Physics';
  String _formBudget = 'Under \$1,000/year (Requires scholarships/free paths)';
  String _formInternet = 'Mobile Data only, capped (Low Bandwidth)';
  String _formLifestyle = 'Remote Freelancer or Tech Entrepreneur';

  // ── Getters ────────────────────────────────────────────────────
  UserProfile get userProfile => _userProfile;
  List<ChatMessage> get chatMessages => _chatMessages;
  CareerSimulation? get simulation => _simulation;
  String get currentScreen => _currentScreen;
  bool get isSimulating => _isSimulating;
  bool get isChatLoading => _isChatLoading;
  bool get simulationComplete => _simulationComplete;

  int get formAge => _formAge;
  String get formCountry => _formCountry;
  String get formInterests => _formInterests;
  String get formSkills => _formSkills;
  String get formGrades => _formGrades;
  String get formBudget => _formBudget;
  String get formInternet => _formInternet;
  String get formLifestyle => _formLifestyle;

  // ── Init ───────────────────────────────────────────────────────
  Future<void> init() async {
    _userProfile = await _repo.getUserProfile();
    _chatMessages = await _repo.getChatMessages();
    _simulation = await _repo.getSimulation();
    notifyListeners();
  }

  // ── Navigation ─────────────────────────────────────────────────
  void navigateTo(String screen) {
    _currentScreen = screen;
    notifyListeners();
  }

  // ── Form setters ───────────────────────────────────────────────
  void updateFormFields({
    int? age,
    String? country,
    String? interests,
    String? skills,
    String? grades,
    String? budget,
    String? internet,
    String? lifestyle,
  }) {
    if (age != null) _formAge = age;
    if (country != null) _formCountry = country;
    if (interests != null) _formInterests = interests;
    if (skills != null) _formSkills = skills;
    if (grades != null) _formGrades = grades;
    if (budget != null) _formBudget = budget;
    if (internet != null) _formInternet = internet;
    if (lifestyle != null) _formLifestyle = lifestyle;
    notifyListeners();
  }

  // ── Save profile & run simulation ─────────────────────────────
  Future<void> saveUserProfileAndRunSimulation() async {
    _isSimulating = true;
    notifyListeners();

    final profile = UserProfile(
      age: _formAge,
      country: _formCountry,
      interests: _formInterests,
      skills: _formSkills,
      grades: _formGrades,
      budget: _formBudget,
      internetAccess: _formInternet,
      preferredLifestyle: _formLifestyle,
      xp: _userProfile.xp,
      level: _userProfile.level,
      streak: _userProfile.streak,
      futureScore: _userProfile.futureScore,
      isPremium: _userProfile.isPremium,
      favoriteTheme: _userProfile.favoriteTheme,
    );

    await _repo.saveProfile(profile);
    _simulation = await _repo.runCareerSimulation(profile);
    _userProfile = await _repo.getUserProfile();

    _isSimulating = false;
    _simulationComplete = true;
    _currentScreen = 'dashboard';
    notifyListeners();
  }

  // ── Premium toggle ─────────────────────────────────────────────
  Future<void> togglePremium(bool isPremium) async {
    final updated = _userProfile.copyWith(isPremium: isPremium);
    await _repo.saveProfile(updated);
    _userProfile = updated;
    notifyListeners();
  }

  // ── Theme ──────────────────────────────────────────────────────
  Future<void> setFavoriteTheme(String theme) async {
    final updated = _userProfile.copyWith(favoriteTheme: theme);
    await _repo.saveProfile(updated);
    _userProfile = updated;
    notifyListeners();
  }

  // ── Chat ───────────────────────────────────────────────────────
  Future<void> sendChatMessage(String text) async {
    if (text.trim().isEmpty) return;
    _isChatLoading = true;
    notifyListeners();

    await _repo.sendMessage('user', text);
    _chatMessages = await _repo.getChatMessages();
    _userProfile = await _repo.getUserProfile();

    _isChatLoading = false;
    notifyListeners();
  }

  Future<void> clearChatHistory() async {
    await _repo.clearChat();
    _chatMessages = [];
    notifyListeners();
  }

  // ── Daily reward ───────────────────────────────────────────────
  Future<void> claimDailyReward() async {
    const bonus = 25;
    final newXp = _userProfile.xp + bonus;
    final updated = _userProfile.copyWith(
      xp: newXp,
      level: 1 + (newXp ~/ 300),
      streak: _userProfile.streak + 1,
      lastUpdate: DateTime.now().millisecondsSinceEpoch,
    );
    await _repo.saveProfile(updated);
    _userProfile = updated;
    notifyListeners();
  }
}
